### PR TITLE
Implement PUT for /api/token and /api/provider

### PR DESCRIPTION
**Pull request by fredx30**
This mirrors the implementation of the `PUT /api/project` for the new endpoints `PUT /api/token` and `PUT /api/provider`

There is also a missing reference to the variable ApiUrl on Client. This is changed to match the source struct as `APIURL`. 

This goes together with a few other PRs:
- (This) The implementable interface https://github.com/iver-wharf/wharf-api-client-go/pull/8
- The PUT /api/token https://github.com/iver-wharf/wharf-api/pull/26
- The PUT /api/provider https://github.com/iver-wharf/wharf-api/pull/28
- The gtihub-provider changes that require the rest https://github.com/iver-wharf/wharf-provider-github/pull/10

### CODE REVIEW

File: pkg/wharfapi/provider.go [hunk 1]
```diff
@@ -71,15 +72,40 @@ func (c Client) GetProvider(providerName string, urlStr string, uploadURLStr str
...
+
+	apiURL := fmt.Sprintf("%s/api/provider", c.APIURL)
+	ioBody, err := doRequest("PUT | PROVIDER |", http.MethodPut, apiURL, body, c.AuthHeader)
+	if err != nil {
+		return Provider{}, err
+	}
+
+	defer (*ioBody).Close()
+
+	newProvider := Provider{}
```
Comment (jilleJr) [inline pkg/wharfapi/provider.go:91]: no need to initialize these values. More common to just declare them as zero values using the following syntax:

```suggestion
	var newProvider Provider
```

File: pkg/wharfapi/token.go [hunk 2]
```diff
@@ -67,15 +68,40 @@ func (c Client) GetToken(token string, userName string) (Token, error) {
...
+
+	apiURL := fmt.Sprintf("%s/api/token", c.APIURL)
+	ioBody, err := doRequest("PUT | TOKEN |", http.MethodPut, apiURL, body, c.AuthHeader)
+	if err != nil {
+		return Token{}, err
+	}
+
+	defer (*ioBody).Close()
+
+	newToken := Token{}
```
Comment (jilleJr) [inline pkg/wharfapi/token.go:87]: no need to initialize these values. More common to just declare them as zero values using the following syntax:

```suggestion
	var newToken Token
```

File: pkg/wharfapi/provider.go [hunk 3]
```diff
@@ -71,15 +72,40 @@ func (c Client) GetProvider(providerName string, urlStr string, uploadURLStr str
...
+	err = json.NewDecoder(*ioBody).Decode(&newProvider)
+	if err != nil {
+		return Provider{}, err
+	}
```
Comment (jilleJr) [inline pkg/wharfapi/provider.go:95]: nit: As no new variable is declared, this can be inlined

```suggestion
	if err := json.NewDecoder(*ioBody).Decode(&newProvider); err != nil {
		return Provider{}, err
	}
```

File: pkg/wharfapi/token.go [hunk 4]
```diff
@@ -67,15 +68,40 @@ func (c Client) GetToken(token string, userName string) (Token, error) {
...
+	err = json.NewDecoder(*ioBody).Decode(&newToken)
+	if err != nil {
+		return Token{}, err
+	}
```
Comment (jilleJr) [inline pkg/wharfapi/token.go:91]: nit: Same as other comment; this can be inlined

```suggestion
	if err := json.NewDecoder(*ioBody).Decode(&newToken); err != nil {
		return Token{}, err
	}
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -14,6 +14,8 @@ This project tries to follow [SemVer 2.0.0](https://semver.org/).
 
 ## v1.3.0 (WIP)
 
+- Removed field ProviderID from Token
```
Comment (jilleJr) [inline CHANGELOG.md:17]: This changelog note seems unrelated

**Review by jilleJr — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (fredx30) [reply CHANGELOG.md]: Updated with relevant changes for this PR.

**Review by jilleJr — APPROVED** [review]
LGTM!

**Review by jilleJr — APPROVED** [review]

File: pkg/wharfapi/token.go [hunk 6]
```diff
@@ -66,15 +67,39 @@ func (c Client) GetToken(token string, userName string) (Token, error) {
 	return tokens[0], nil
 }
 
+func (c Client) PutToken(token Token) (Token, error) {
+	body, err := json.Marshal(token)
+	if err != nil {
+		return Token{}, err
+	}
+
+	log.WithField("project", string(body)).Traceln()
```
Comment (Pikabanga) [inline pkg/wharfapi/token.go:76]: Is this token sensitive? Maybe it shouldn't be logged.

**Review by Pikabanga — CHANGES_REQUESTED** [review]

File: pkg/wharfapi/token.go (hunk 6 above)
Comment (fredx30) [reply pkg/wharfapi/token.go:76]: Yes i belive so. Im unsure about the logging strategy here though. We dont really log anything from this repo. It appears the trace level of the log never gets output. 
With the descriptions available on the http responses i belive this logging wouldnt do much more good in its current state. I cant think of any more info we would really need/want here so i would advocate for dropping this traceln log from the put method here and in the provider.

**Review by Pikabanga — APPROVED** [review]
